src/main/java/com/google/enterprise/cloudsearch/sharepoint/SharePointConfiguration.java

```diff
@@ -88,8 +86,7 @@ private SharePointConfiguration(Builder builder) throws URISyntaxException {
       this.siteCollectionOnly = builder.sharePointUrl.getUrl().split("/").length > 3;
     }
 
-    this.siteCollectionsToInclude =
-        Collections.unmodifiableSet(new HashSet<>(builder.siteCollectionsToInclude));
+    this.siteCollectionsToInclude = builder.siteCollectionsToInclude;
```

Comment:
If you're doing this, you need to change the field itself, so that this is more obviously safe (and the later cast is not needed).